CRM/Civicase/Hook/alterMailParams/SubjectCaseTypeCategoryProcessor.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+
+/**
+ * Class CRM_Civicase_Hook_alterMailParams_CaseTypeCategorySubjectProcessor.
+ */
+class CRM_Civicase_Hook_alterMailParams_SubjectCaseTypeCategoryProcessor {
+
+  /**
+   * A substring of email subject that should be replaced.
+   *
+   * @var string
+   *   This substring will be replaced with respective case type category name.
+   */
+  private static $toReplace = '[case ';
+
+  /**
+   * Email subject processor.
+   *
+   * Replaces word 'case' with related case type category in email subject.
+   *
+   * @param array &$params
+   *   Mail parameters.
+   * @param string $context
+   *   Mail context.
+   * @param int $caseId
+   *   Case id.
+   */
+  public static function run(array &$params, $context, $caseId) {
+    if (!self::shouldRun($params, $context, $caseId)) {
```

Comment:
`shouldRun` should not be called statically once this function is made not to be static.